code/datums/config/config_num.dm

```diff
@@ -0,0 +1,454 @@
+/decl/config/num
+	abstract_type = /decl/config/num
+	config_flags = CONFIG_FLAG_NUM | CONFIG_FLAG_HAS_VALUE
+	default_value = 0
+	var/min_value = -(INFINITY)
+	var/max_value = INFINITY
+
+/decl/config/num/sanitize_value()
+	value = clamp(value, min_value, max_value)
```

Comment:
I'd add rounding here too. Since sometimes you want strictly integers or allow decimals.

Reply:
What I did was not make a flag, but a rounding value like this:
```
/decl/config
   var/rounding = 0.01

/decl/config/proc/parse(val)
   return round(clamp(val, minimum, maximum), rounding)

/decl/config/someinteger
   rounding = null
```

So you can just set the rounding value to something else if you need/expect more precision and also how much precision.
And after having gone through a lot of the numeric config entry, most of them seems to be about strictly integers? Or at least they wouldn't produce the expected result if they had a decimal part.